Source/SwiftLintFramework/Rules/RuleConfigurations/CyclomaticComplexityConfiguration.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  CyclomaticComplexityConfiguration.swift
+//  SwiftLint
+//
+//  Created by Mike Welles on 2/9/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+import Foundation
+import SourceKittenFramework
+
+private enum ConfigurationKey: String {
+    case warning = "warning"
+    case error = "error"
+    case ignoresCaseStatements = "ignores_case_statements"
+
+    static func all() -> [ConfigurationKey] {
+        return [
+            .warning,
+            .error,
+            .ignoresCaseStatements
+        ]
+    }
+    static func allValues() -> [String] {
+        return all().map { $0.rawValue }
+    }
+}
+
+public struct CyclomaticComplexityConfiguration: RuleConfiguration, Equatable {
+    public var consoleDescription: String {
+        return length.consoleDescription + ", ignores switch statements: \(ignoresCaseStatements)"
+    }
+
+    public static let defaultComplexityStatements: Set<StatementKind> = Set([
+        .forEach,
+        .if,
+        .guard,
+        .for,
+        .repeatWhile,
+        .while,
+        .case
+        ])
+
+    var length: SeverityLevelsConfiguration
+
+    private(set) public var complexityStatements: Set<StatementKind>
+
+    var ignoresCaseStatements: Bool {
+        didSet {
+            if ignoresCaseStatements {
+                complexityStatements.remove(.case)
+            } else {
+                complexityStatements.insert(.case)
+            }
+        }
+    }
+
+    var params: [RuleParameter<Int>] {
```

Comment:
@jpsim  -- Following up on this, since it's the one remaining issue.    I can't add a set to this accessor, it's read-only -- returns length.params, which is itself read-only.